src/backend/catalog/namespace.c

```diff
@@ -1982,6 +1982,13 @@ recomputeNamespacePath(void)
 			elog(DEBUG3, "recompute search_path[%s] when acl_type is ranger", namespace_search_path);
 		}
 	}
+	else 
+	{
+		if (aclType == HAWQ_ACL_RANGER && debug_query_string != NULL)
+		{
+			last_query_sign = string_hash(debug_query_string, strlen(debug_query_string));
+		}
+	}
```

Comment:
How about set current_qurey_sign before the if statement 'if (namespaceSearchPathValid && namespaceUser == roleid)' and set last_query_sign after the if statement? Can this make the return condition more clear?